pkg/metallb/frr.go

```diff
@@ -0,0 +1,664 @@
+package metallb
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"strconv"
+	"time"
+
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+
+	"github.com/golang/glog"
+	"github.com/openshift-kni/eco-goinfra/pkg/clients"
+	"github.com/openshift-kni/eco-goinfra/pkg/msg"
+	frrtypes "github.com/openshift-kni/eco-goinfra/pkg/schemes/metallb/frrtypes"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// FrrConfigurationBuilder provides struct for the FrrConfiguration object containing connection to
+// the cluster and the FrrConfiguration definitions.
+type FrrConfigurationBuilder struct {
+	Definition *frrtypes.FRRConfiguration
+	Object     *frrtypes.FRRConfiguration
+	apiClient  *clients.Settings
+	errorMsg   string
+}
+
+// NewFrrConfigurationBuilder creates a new instance of FRRConfiguration.
+func NewFrrConfigurationBuilder(
+	apiClient *clients.Settings, name, nsname string) *FrrConfigurationBuilder {
+	glog.V(100).Infof(
+		"Initializing new Frrconfiguration structure with the following params: %s, %s",
+		name, nsname)
+
+	builder := FrrConfigurationBuilder{
+		apiClient: apiClient,
+		Definition: &frrtypes.FRRConfiguration{
+			TypeMeta: metav1.TypeMeta{
+				Kind:       FrrConfigurationKind,
+				APIVersion: fmt.Sprintf("%s/%s", FRRAPIGroup, APIVersion),
+			},
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      name,
+				Namespace: nsname,
+			}, Spec: frrtypes.FRRConfigurationSpec{
+				BGP: frrtypes.BGPConfig{},
+			},
```

Comment:
This is the zero value for spec and can be left out too

Reply:
removed